### DISCUSSION TITLE
Securing Data

### DISCUSSION THREAD

**Discussion by emik** [prisma/prisma#8673] [Q&A] [answered]
**EDIT**: Looks like this is a [whole discussion](https://github.com/prisma/prisma/discussions/2846) around what is actually called multiple tenancy that has already occurred for prisma and there is an [open issue ](https://github.com/prisma/prisma/issues/2077)for it. [prisma-multi-tenant](https://github.com/errorname/prisma-multi-tenant) also exists as a potential solution, though upon looking at the github issues it seems it is not compatible with latest prisma versions...

Hello!

Is there a general convention for restricting access to data in Prisma (using MySQL currently)? I have multiple organisations in my database and would like to ensure that clients can only view data associated with their organisation id when fetching from the API using a JWT. It seems to me that I can either:
- Use one unique database per organisation
- Manually require the organisation id embedded in the validated JWT to be passed into every single query's `where`
- Another solution I thought may be to ensure `organisationId` is applied on every table and check with middleware that there is no id mismatch with any of the returned data. Though this doesn't prevent updates/deletions.

I've searched the docs but can't find anything that relevant, is there a preferred way of doing this using Prisma? All of these seem a bit cumbersome.

Thanks for any insights you may provide!

**Comment by ryands17** [comment, marked as answer]
@emik 👋 

The option with Middleware prevent for all operations including updates and deletes. The `organisationId` can be checked via the user's credentials passed for the request and then you could append that in the middleware.

**Reply by emik** [reply]
Thanks! I'm going to pursue this option I think :)

**Reply by ryands17** [reply]
Sure, let me know how it goes and/or if you face any issues 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
